Src/Newtonsoft.Json/JsonReader.cs

```diff
@@ -1109,5 +987,50 @@ public virtual void Close()
             _tokenType = JsonToken.None;
             _value = null;
         }
+
+        internal void ReadAndAssert()
+        {
+            if (!Read())
+            {
+                throw JsonSerializationException.Create(this, "Unexpected end.");
+            }
+        }
+
+        internal bool ReadAndSkipComments()
+        {
+            return Read() && SkipComments();
+        }
+
+        internal bool SkipComments()
```

Comment:
XmlReader has a method called MoveToContent that fulfills the same purpose. I think SkipComments could be called ReadToContent.